src/main/java/org/team1540/base/power/PowerManager.java

```diff
@@ -122,23 +151,44 @@ private void stopScaling() {
   }
 
 
+  /**
+   * Determines if the current is currently spiking. If power limiting is not engaged,
+   * returns pdp.getTotalCurrent() &gt; currentSpikePeak.
+   * If power limiting is engaged, returns pdp.getTotalCurrent() &gt; currentTarget - currentMargin.
+   *
+   * @return Boolean representing if the current is spiking.
+   */
+  public boolean currentIsSpiking() {
+    if (!currentTimeHasPassed()) {
+      return pdp.getTotalCurrent() > currentSpikePeak;
+    } else {
+      return pdp.getTotalCurrent() > currentTarget - currentMargin;
+    }
+  }
+
   /**
    * Determines if the voltage is currently spiking. If power limiting is not engaged,
-   * returns pdp.getTotalCurrent() &gt; spikePeak. If power limiting is engaged, returns
-   * target - pdp.getTotalCurrent() &gt; margin.
+   * returns RobotController.getBatteryVoltage() &lt; voltageDipLow || RobotController.isBrownedOut();
+   * If power limiting is engaged, returns pdp.getVoltage() &lt; voltageDipLow + voltageMargin ||
+   * RobotController.isBrownedOut();.
    *
    * @return Boolean representing if the voltage is spiking.
    */
-  public boolean isSpiking() {
-    if (!timeHasPassed()) {
-      return pdp.getTotalCurrent() > spikePeak;
+  public boolean voltageIsDipping() {
+    if (!voltageTimeHasPassed()) {
+      return RobotController.getBatteryVoltage() < voltageDipLow || RobotController.isBrownedOut();
     } else {
-      return pdp.getTotalCurrent() > target - margin;
+      return RobotController.getBatteryVoltage() < voltageDipLow + voltageMargin || RobotController
+          .isBrownedOut();
     }
   }
 
-  private boolean timeHasPassed() {
-    return (theTimer.get() > spikeLength);
+  private boolean currentTimeHasPassed() {
```

Comment:
Same here.

Reply:
Would this be timeHasPassedCurrent()?

Reply:
Methods should start with a verb, so either `hasCurrentTimePassed()` or refactor to something like `isCurrentSpiking()` (which also might make it easier to read)

Reply:
isCurrentSpiking is a different method tho...
Also, the style guide does technically say "typically", not always.

Reply:
Fair, didn't notice that. `hasCurrentTimePassed()`, then.

Reply:
I'm putting current at the end, since it's "Has the time passed for the current?" now "Has the current passed its time?" Also the current in the middle makes you think of the other definition of current.

Reply:
Fair.